lib/features/children/details/cubit/child_details_cubit.dart

```diff
@@ -77,13 +78,67 @@ class ChildDetailsCubit extends Cubit<ChildDetailsState> {
     }
   }
 
+  Future<void> topUp(int amount) async {
+    await _logTopUpEvent(amount);
```

Comment:
Small thing. Personally i would not "await" analytics event. So I would always wrap them in unawaited(); My reasoning is we wouldn't want to wait on analytics in the user experience and if analytics fail its not blocking for the app.